src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php

```diff
@@ -0,0 +1,166 @@
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressException;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Address\Query\GetManufacturerAddressForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Address\QueryResult\EditableManufacturerAddress;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerConstraintException;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Builder\FormBuilderInterface;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Handler\FormHandlerInterface;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Is responsible for "Sell > Catalog > Brands & Suppliers" page.
+ */
+class ManufacturerController extends FrameworkBundleAdminController
+{
+    /**
+     * Show & process address creation.
+     *
+     * @AdminSecurity("is_granted('create', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAddressAction(Request $request)
+    {
+        $addressFormBuilder = $this->getAddressFormBuilder();
+        $addressFormHandler = $this->getAddressFormHandler();
+        $addressForm = $addressFormBuilder->getForm();
+        $addressForm->handleRequest($request);
+
+        try {
+            $result = $addressFormHandler->handle($addressForm);
+
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_manufacturers_index');
+            }
+        } catch (DomainException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+
+            if ($e instanceof ManufacturerConstraintException) {
+                return $this->redirectToRoute('admin_manufacturers_index');
+            }
+        }
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/Address/create.html.twig', [
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('Add new address', 'Admin.Orderscustomers.Feature'),
+            'addressForm' => $addressForm->createView(),
+        ]);
+    }
+
+    /**
+     * Show & process address editing.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $addressId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAddressAction(Request $request, $addressId)
+    {
+        $addressId = (int) $addressId;
```

Comment:
either this line or further int casting of `$addressId` is redundant in this action